### PR TITLE
Ability to set request timeout, made opts parameters optional

### PR DESCRIPTION
**Pull request by DaWe35**
This PR includes two backward-compatible updates.
- Ability to set the request timeout
- The ability to NOT to set all opts parameters.
  Previously if you wanted to pass the `opts` parameter, it was mandatory to declare all variables in the `opts` object. Now you can declare the variables that are important for you, and leave the rest default.
  Example:
  ``` python
  opts = type('obj', (object,), {
  	'portal_url': portal,
  	'timeout': 20
  })
  skylink = Skynet.upload_file('2MBfile.txt', opts)
   ```

Complete example for upload [(source)](https://github.com/DaWe35/SkyLive/blob/master/benchmark_portals.py):

``` python
opts = type('obj', (object,), {
	'portal_url': portal,
	'timeout': 20
})
try:
	try:
		skylink = Skynet.upload_file('file.txt', opts)
		print(skylink)
	except TimeoutError as e:
		# timed out

except Exception as e:
	# other error
```

### CODE REVIEW

File: siaskynet/skynet.py [hunk 1]
```diff
@@ -12,19 +12,55 @@ def uri_skynet_prefix():
...
+            'portal_upload_path': portal_upload_path,
+            'portal_file_fieldname': portal_file_fieldname,
+            'portal_directory_file_fieldname': portal_directory_file_fieldname,
+            'custom_filename': custom_filename,
+            'timeout': timeout
         })
 
     @staticmethod
-    def default_download_options():
+    def fill_with_default_download_options(opts):
```
Comment (m-cat) [inline siaskynet/skynet.py:56]: This is missing setting the `timeout` field

**Comment by marcinja** [comment]
Looks good to me. Just needs the changes m-cat already mentioned.

**Comment by DaWe35** [comment]
@m-cat 1. What about making `default_upload_options` and `default_download_options` able to handle deficient `opts`? I mean do we really need the `fill_with...` functions?
2. agree

**Comment by m-cat** [comment]
> @m-cat 1. What about making `default_upload_options` and `default_download_options` able to handle deficient `opts`? I mean do we really need the `fill_with...` functions?

Interesting idea. In my opinion the `default_` functions should just return the default `opts` as before. I don't see the reason for a user to pass in a deficient `opts`, the upload and download methods will just fill in the missing `opts` anyway. I would keep the `default_` functions simple, behaving the same way as in the other SDKs, while we make the `fill_with` functions private. This way we don't complicate the API at all and still have elegant option-filling internally.

**Comment by DaWe35** [comment]
@m-cat I cant call private functions while the because there is no instantiated class instance here.

```
    @staticmethod
    def default_upload_options(self):
        return self.__fill_with_default_upload_options()
```

**Comment by m-cat** [comment]
@DaWe35 Since these are static methods there is no class instance, you need to do `Skynet.fill_with_default_upload_options()`

**Comment by m-cat** [comment]
Okay, I've seen you've done that. Looks great to me! Will give it one more look-through.

**Pull request merged**